lib/templ.ml

```diff
@@ -210,19 +268,25 @@ and eval_simple_variable conf =
   | "suffix" ->
     (* On supprime de env toutes les paires qui sont dans (henv @ senv) *)
     let l = List.fold_left (fun accu (k, _) -> List.remove_assoc k accu) conf.env (List.rev_append conf.henv conf.senv) in
-    List.fold_left (fun c (k, v) ->
-      let v = Adef.as_string v in
-      if ( ( k = "oc" || k = "ocz" ) && v = "0" ) || k = "" then c
-      else c ^ k ^ "=" ^ v ^ "&")
-    "" l
+    let s = List.fold_left (fun c (k, v) ->
+        let v = Adef.as_string v in
+        if ( ( k = "oc" || k = "ocz" ) && v = "0" ) || k = "" then c
+        else c ^ k ^ "=" ^ v ^ "&")
+      "" l
+    in
+    if s.[String.length s - 1] = '&' then String.sub s 0 (String.length s - 1)
+    else s (* suppress last & *)
   | "url" ->
       let c :> string = Util.commd conf in
       (* On supprime de env toutes les paires qui sont dans (henv @ senv) *)
       let l :> (string * string) list =
         List.fold_left (fun accu (k, _) -> List.remove_assoc k accu) conf.env
           (List.rev_append conf.henv conf.senv)
       in
-      List.fold_left (fun c (k, v) -> c ^ k ^ "=" ^ v ^ "&") c l
+      let u = List.fold_left (fun c (k, v) -> c ^ k ^ "=" ^ v ^ "&") c l in
+      if u.[String.length u - 1] = '&' then String.sub u 0 (String.length u - 1)
```

Comment:
this can fail too. (if `l = []` then `u = ""`)

Reply:
Bien vu! Merci